python/pyspark/streaming/tests.py

```diff
@@ -1636,7 +1636,13 @@ def search_kinesis_asl_assembly_jar():
         jars = "%s,%s,%s,%s,%s" % (kafka_assembly_jar, flume_assembly_jar, mqtt_assembly_jar,
                                    mqtt_test_jar, kinesis_asl_assembly_jar)
 
-    os.environ["PYSPARK_SUBMIT_ARGS"] = "--jars %s pyspark-shell" % jars
+    # We need to set userClassPathFirst here because the streaming data source classes are also
```

Comment:
Hmm, maybe I should just remove those classes from the list in `AbstractCommandBuilder` and have that only include the classes that would have been in the assembly JAR in the first place.